pkg/resources/pilot/deployment.go

```diff
@@ -157,6 +157,20 @@ func (r *Reconciler) containers() []apiv1.Container {
 		})
 	}
 
+	if util.PointerToBool(r.Config.Spec.Pilot.EnableProtocolSniffingOutbound) {
+		discoveryContainer.Env = append(discoveryContainer.Env, apiv1.EnvVar{
+			Name:  "PILOT_ENABLE_PROTOCOL_SNIFFING_FOR_OUTBOUND",
+			Value: "true",
```

Comment:
Are u sure we always want to set this to `true`? If `util.PointerToBool(r.Config.Spec.Pilot.EnableProtocolSniffingOutbound)` is `false` then we don't set this env var, but won't it be initialized to be `true` anyway: https://github.com/istio/istio/blob/6d0953f791f76f614c2478f463eef166ea399105/pilot/pkg/features/pilot.go#L177?

Reply:
👍